lib/BloqadeExpr/src/types.jl

```diff
@@ -164,7 +164,7 @@ Type for Rydberg interactive term.
 - `C`: the interaction strength, default unit is `MHz⋅μm^6`. default value is `2π * 862690 * MHz*µm^6`.
 """
 Base.@kwdef struct RydInteract{D} <: AbstractTerm{D}
-    atoms::Vector
+    atoms::Union{Vector,BoundedLattice}
```

Comment:
just wondering what interface does `BoundedLattice` share with a `Vector`? if we are expecting putting the lattice inside along with a vector. Otherwise we should think to remove the `Vector` type specification to keep the behaviour consistent.